src/codeblockui.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module code-block/codeblockui
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import Model from '@ckeditor/ckeditor5-ui/src/model';
+import SplitButtonView from '@ckeditor/ckeditor5-ui/src/dropdown/button/splitbuttonview';
+import { createDropdown, addListToDropdown } from '@ckeditor/ckeditor5-ui/src/dropdown/utils';
+import { getLocalizedLanguageDefinitions } from './utils';
+
+import codeBlockIcon from '../theme/icons/codeblock.svg';
+import '../theme/codeblock.css';
+
+/**
+ * The code block UI plugin.
+ *
+ * Introduces the `'codeBlock'` dropdown.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class CodeBlockUI extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const t = editor.t;
+		const componentFactory = editor.ui.componentFactory;
+		const localizedLanguageDefinitions = getLocalizedLanguageDefinitions( editor );
+
+		componentFactory.add( 'codeBlock', locale => {
+			const command = editor.commands.get( 'codeBlock' );
+			const dropdownView = createDropdown( locale, SplitButtonView );
+			const splitButtonView = dropdownView.buttonView;
+
+			splitButtonView.set( {
+				label: t( 'Insert code block' ),
+				tooltip: true,
+				icon: codeBlockIcon,
+				isToggleable: true
+			} );
+
+			splitButtonView.bind( 'isOn' ).to( command, 'value', value => !!value );
+
+			splitButtonView.on( 'execute', () => {
+				editor.execute( 'codeBlock', {
+					language: localizedLanguageDefinitions[ 0 ].class,
```

Comment:
I'm missing here why we use `[ 0 ]` here. The information about this behaviour may be forgotten in time. I'd propose to use either a comment or a variable (e.g. `defaultDefinition`) to denote why we're using this.